### PR TITLE
Set tags in alphabetical order, emojis first

### PR DESCRIPTION
**Pull request by werner33**
## Describe your changes

Feel free to disregard if someone else is going to open a request on this issue. I teach software development and we used this issue as a lesson in class today. 

I've put the tags in alphabetical order, ignoring emojis. I've refactored the file with more expressive names and moved everything inside a function. I've added an interface for the resources being imported. 

## Related issue number/link

Fixes https://github.com/ClimateTown/knowledge-hub/issues/129

Thanks for contributing! 🥳🚀🌳

### CODE REVIEW

**Comment by VeckoTheGecko** [comment]
Thanks for the PR, and its great that it was a good teaching example! I'll be able to review this PR this weekend when I have more time.

Thanks @Micahg05 for #142 as well. I'll look at it as well this weekend and compare approaches. I'll be working on documenting more issues over the next week so that the direction of the project is clearer, and there is more space to avoid people doubling up on issues.

File: src/routes/resources/+page.server.ts [hunk 1]
```diff
@@ -2,28 +2,54 @@ import type { PageServerLoad } from './$types'
...
-tags = [...new Set(tags)]
 
-//Sorting resources by newest (assuming newest is at the bottom of the file )
-const yml_data_sort = [...yml_data].reverse();
+const sortAlphabeticallyIgnoringEmojis = (a: string, b: string) => {
+  const aWithoutEmojis = removeEmojisFromStr(a);
+  const bWithoutEmojis = removeEmojisFromStr(b);
+
+  return aWithoutEmojis.localeCompare(bWithoutEmojis);
+}
```
Comment (VeckoTheGecko) [inline src/routes/resources/+page.server.ts:37]: This isn't the goal of the sorting. From #142:

> Alphabetic sorting of tags in resources filter (for easier grouping). *Tags with emojis should be at the top of the list.*

We would like to prioritise the emoji tags which are special tags (and also flag particularly high quality resources).

**Review by VeckoTheGecko — COMMENTED** [review]
Thanks again for this PR and the refactor. A bit of a misunderstanding surrounding the issue. Would you have time/be willing to remedy this @werner33? If not, we can open continuing the PR to another member of the community.

**Comment by VeckoTheGecko** [comment]
@all-contributors please add @werner33 for code

File: src/routes/resources/+page.server.ts (hunk 1 above)
Comment (werner33) [reply src/routes/resources/+page.server.ts:37]: I'll fix this today, no problem. 

Thanks.

**Comment by werner33** [comment]
Closing as I work out some more changes. Will reopen later.

**Comment by VeckoTheGecko** [comment]
Feel free to discuss implementation/changes here.

I see that you're looking at adding scripts to validate resource tags? This would be a welcome change, but I think it would be much easier to use the existing `jsonschema` test which validates that the `data/resources.yml` adheres to [`data/resources.schema.json`](https://github.com/ClimateTown/knowledge-hub/blob/main/data/resources.schema.json)

https://github.com/ClimateTown/knowledge-hub/blob/main/data/yaml_test.py

Adding a list of approved tags is just a matter of modifying the schema to add an `enum` field, and then the test will handle the rest with rich error messages.

```diff
...
            "tags": {
                "description": "A list of tags to categorize the resource.",
                "type": "array",
                "items": {
                    "type": "string",
+                   "enum": ["tag1", "tag2", "tag3"]  // Add your allowed tags here
                },
                "minItems": 1,
                "maxItems": 6
            }
...
```

Regarding the sorting of the tags, it can be done by moidifying your prior implementation:

- Split tags into two lists, one with emojis one without
- Sort both lists ignoring emojis
- Join lists back `emojisList + nonEmojisList`

**Pull request closed without merging**